doc/Migration_guide_0.4.md

```diff
@@ -0,0 +1,93 @@
+# Migrating Java service to v0.4
+
+## Update the client apps
+Update the clients to submit transactions in the new format, introduced in Exonum 0.10.
+The most notable change in the format is that the public key of the transaction author 
+is always included in the transaction message. This allows the framework to verify 
+the message signature with no conversion to an executable transaction.
+
+If the client application is in Java, use `com.exonum.binding.common.message.TransactionMessage`
+from "[com.exonum.binding:exonum-java-binding-common:0.4][common-0.4]";
```

Comment:
We might as well release `com.exonum.client:exonum-light-client:0.1` by that time :thinking: